### PR TITLE
Improve searching for symbols when a name is known ahead of time. (And avoid O(n) enumerations)

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/26325.

This also switches the backing store for the MemberNames collection to an ImmutableHashSet.  That has two benefits.  One, it fixes https://github.com/dotnet/roslyn/issues/26328 where the current system allows a client to potentially corrupt internal compiler state.

Second, it allows for O(log(n)) name lookups instead of O(n).  this is because we use ImmutableHashSet which is an efficient AVL tree where the hashcode of the object is the key for the node**.  This is an improvement on top of our own SmallDict ** , and means we can be immutable, have fast lookups, and not have much more overhead than an array/immutablearray.

--

**  SmallDictionary is effectively the same, just with space for a 'value'.  As we don't need that here, we save by just using ImmutableHashSet.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @VSadov @jcouv

**Comment by gafter** [comment]
This appears to contain deltas from #26325. Can we wrap that one up before I review this?

**Comment by CyrusNajmabadi** [comment]
@gafter Yup.  As i mentioned:

> Followup to #26325.

I'll ping you when that goes in and this is ready to review.

**Comment by CyrusNajmabadi** [comment]
@gafter Dependent PR has been merged.  This is ready for review now.  Also @jcouv.

**Comment by jcouv** [comment]
Yup. Removed "Blocked" tag. Will take a look tonight or tomorrow.

**Comment by CyrusNajmabadi** [comment]
@gafter @jcouv can you take a peek plz?   Thanks!

File: src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb [hunk 1]
```diff
@@ -549,12 +547,28 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return children.ToImmutableAndFree()
         End Function
 
-        Private Function GetNonTypeMemberNames(members As SyntaxList(Of StatementSyntax), ByRef declFlags As SingleTypeDeclaration.TypeDeclarationFlags) As String()
+        ''' <summary>
+        ''' Pool of builders used to create our member name sets.  Importantly, these use 
+        ''' <see cref="CaseInsensitiveComparison.Comparer"/> so that name lookup happens in an
+        ''' appropriate manner for VB identifiers. This allows fast member name O(log(n)) even if
+        ''' the casing doesn't match.
+        ''' </summary>
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb]: note: we have lot of case sensitivty tests for VB (including directly in symbol search) to validate this behavior.

**Comment by CyrusNajmabadi** [comment]
Compiling roslyn shows no change in perf for me.  How do you guys normally validate compiler changes wrt perf yourself?

**Comment by jcouv** [comment]
I'll let @agocke reply (he has some instructions on how to validate).

**Comment by CyrusNajmabadi** [comment]
Awesome. I'll try this out this weekend!

**Comment by CyrusNajmabadi** [comment]
@agocke I tried this, but i got:

> C:\GitHub\roslyn-internal\roslyn-tools\src\CompilerPerfTests [master ≡]> .\run-perf.ps1
The term 'Ensure-SdkInPath' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

**Comment by agocke** [comment]
@CyrusNajmabadi Fixed here: https://github.com/dotnet/roslyn-tools/pull/248. Looks like some changes were made to Roslyn that needed to be reflected in the powershell script.

**Comment by CyrusNajmabadi** [comment]
Thanks!  Here are my results:

```
// * Summary *

BenchmarkDotNet=v0.10.9, OS=Windows 10.0.16299
Processor=Intel Core i7-7600U CPU 2.80GHz (Kaby Lake), ProcessorCount=4
Frequency=2835939 Hz, Resolution=352.6169 ns, Timer=TSC
.NET Core SDK=2.1.105
  [Host] : .NET Core 2.0.7 (Framework 4.6.26328.01), 64bit RyuJIT DEBUG

Toolchain=Perf.ExternalProcessToolchain  LaunchCount=0  RunStrategy=Monitoring
TargetCount=25  WarmupCount=0

            Method | Commit |    Mean |    Error |   StdDev | Rank |
------------------ |------- |--------:|---------:|---------:|-----:|
 PlaceholderMethod |   HEAD | 4.907 s | 0.1695 s | 0.2263 s |    1 |
 PlaceholderMethod |  HEAD^ | 4.885 s | 0.1740 s | 0.2322 s |    1 |
```

<details>
// ***** BenchmarkRunner: Start   *****
// Found benchmarks:
//   PlaceholderBenchmarkRunner.PlaceholderMethod: Job-EUGFVE(Toolchain=Perf.ExternalProcessToolchain, LaunchCount=0, RunStrategy=Monitoring, TargetCount=25, WarmupCount=0) [Commit=HEAD^]
//   PlaceholderBenchmarkRunner.PlaceholderMethod: Job-EUGFVE(Toolchain=Perf.ExternalProcessToolchain, LaunchCount=0, RunStrategy=Monitoring, TargetCount=25, WarmupCount=0) [Commit=HEAD]

// Validating benchmarks:
No exporters defined, results will not be persisted.
// **************************
// Benchmark: PlaceholderBenchmarkRunner.PlaceholderMethod: Job-EUGFVE(Toolchain=Perf.ExternalProcessToolchain, LaunchCount=0, RunStrategy=Monitoring, TargetCount=25, WarmupCount=0) [Commit=HEAD^]
// *** Generate ***
// Result = Success
// BinariesDirectoryPath =

// *** Build ***
Building commit: HEAD^

commit b6d9b18913b6f66a2a4759bb8c5a6d3c039bdd88
Author: Cyrus Najmabadi <cyrus.najmabadi@gmail.com>

    Add comments

Repo Dir C:\GitHub\roslyn-internal\roslyn
Binaries Dir C:\GitHub\roslyn-internal\roslyn\Binaries
Running restore
Restore using dotnet at C:\GitHub\roslyn-internal\roslyn\Binaries\Tools\dotnet\dotnet.exe
Restoring Roslyn Toolset
Restoring Roslyn
Microsoft (R) Build Engine version 15.7.66.2115 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\GitHub\roslyn-internal\roslyn\src\Compilers\Core\Portable\CodeAnalysis.csproj...
  Restoring packages for C:\GitHub\roslyn-internal\roslyn\src\Compilers\CSharp\Portable\CSharpCodeAnalysis.csproj...
  Restoring packages for C:\GitHub\roslyn-internal\roslyn\src\Compilers\CSharp\csc\csc.csproj...
  Restore completed in 240.65 ms for C:\GitHub\roslyn-internal\roslyn\src\Compilers\Core\Portable\CodeAnalysis.csproj.
  Restore completed in 240.65 ms for C:\GitHub\roslyn-internal\roslyn\src\Compilers\CSharp\Portable\CSharpCodeAnalysis.csproj.
  Restore completed in 540.87 ms for C:\GitHub\roslyn-internal\roslyn\src\Compilers\CSharp\csc\csc.csproj.
  CodeAnalysis -> C:\GitHub\roslyn-internal\roslyn\Binaries\Release\Dlls\CodeAnalysis\Microsoft.CodeAnalysis.dll
  CSharpCodeAnalysis -> C:\GitHub\roslyn-internal\roslyn\Binaries\Release\Dlls\CSharpCodeAnalysis\Microsoft.CodeAnalysis.CSharp.dll
  csc -> C:\GitHub\roslyn-internal\roslyn\Binaries\Release\Exes\csc\netcoreapp2.0\csc.dll
  csc -> C:\GitHub\roslyn-internal\roslyn\Binaries\Release\Exes\csc\net46\csc.exe

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:26.43
// Result = Success

// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet "C:\GitHub\roslyn-internal\roslyn\Binaries/Release/Exes/csc/netcoreapp2.0/csc.dll" -noconfig @repro.rsp
Result  0: 1 op, 5018241929.75 ns, 5.0182 s/op
Result  1: 1 op, 4836735557.43 ns, 4.8367 s/op
Result  2: 1 op, 4780063322.94 ns, 4.7801 s/op
Result  3: 1 op, 4881035170.36 ns, 4.8810 s/op
Result  4: 1 op, 4738114959.45 ns, 4.7381 s/op
Result  5: 1 op, 4754514818.55 ns, 4.7545 s/op
Result  6: 1 op, 4922997638.52 ns, 4.9230 s/op
Result  7: 1 op, 4708330468.32 ns, 4.7083 s/op
Result  8: 1 op, 4763850350.80 ns, 4.7639 s/op
Result  9: 1 op, 5017143175.51 ns, 5.0171 s/op
Result 10: 1 op, 4747376442.16 ns, 4.7474 s/op
Result 11: 1 op, 4789376640.33 ns, 4.7894 s/op
Result 12: 1 op, 4951771177.02 ns, 4.9518 s/op
Result 13: 1 op, 4657641084.66 ns, 4.6576 s/op
Result 14: 1 op, 4716497428.19 ns, 4.7165 s/op
Result 15: 1 op, 4747601764.35 ns, 4.7476 s/op
Result 16: 1 op, 4824975431.42 ns, 4.8250 s/op
Result 17: 1 op, 5029086309.68 ns, 5.0291 s/op
Result 18: 1 op, 5072212060.98 ns, 5.0722 s/op
Result 19: 1 op, 5802681228.33 ns, 5.8027 s/op
Result 20: 1 op, 5119816046.82 ns, 5.1198 s/op
Result 21: 1 op, 4804410108.96 ns, 4.8044 s/op
Result 22: 1 op, 4913025985.40 ns, 4.9130 s/op
Result 23: 1 op, 4631491368.47 ns, 4.6315 s/op
Result 24: 1 op, 4890601314.06 ns, 4.8906 s/op

Mean = 4.8848 s, StdErr = 0.0464 s (0.95%); N = 25, StdDev = 0.2322 s
Min = 4.6315 s, Q1 = 4.7475 s, Median = 4.8250 s, Q3 = 4.9845 s, Max = 5.8027 s
IQR = 0.2370 s, LowerFence = 4.3920 s, UpperFence = 5.3399 s
ConfidenceInterval = [4.7108 s; 5.0587 s] (CI 99.9%), Margin = 0.1740 s (3.56% of Mean)
Skewness = 2.39, Kurtosis = 9.98


// **************************
// Benchmark: PlaceholderBenchmarkRunner.PlaceholderMethod: Job-EUGFVE(Toolchain=Perf.ExternalProcessToolchain, LaunchCount=0, RunStrategy=Monitoring, TargetCount=25, WarmupCount=0) [Commit=HEAD]
// *** Generate ***
// Result = Success
// BinariesDirectoryPath =

// *** Build ***
Building commit: HEAD

commit bbb77e5df962217e2f8bd52fb7d83600729e4c5e (HEAD -> symbolSearchO1, CyrusNajmabadi/symbolSearchO1)
Merge: b6d9b18913 d4c65895da
Author: Cyrus Najmabadi <cyrus.najmabadi@gmail.com>

    Merge remote-tracking branch 'upstream/master' into symbolSearchO1

Repo Dir C:\GitHub\roslyn-internal\roslyn
Binaries Dir C:\GitHub\roslyn-internal\roslyn\Binaries
Running restore
Restore using dotnet at C:\GitHub\roslyn-internal\roslyn\Binaries\Tools\dotnet\dotnet.exe
Restoring Roslyn Toolset
Restoring Roslyn
Microsoft (R) Build Engine version 15.7.66.2115 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\GitHub\roslyn-internal\roslyn\src\Compilers\Core\Portable\CodeAnalysis.csproj...
  Restoring packages for C:\GitHub\roslyn-internal\roslyn\src\Compilers\CSharp\Portable\CSharpCodeAnalysis.csproj...
  Restoring packages for C:\GitHub\roslyn-internal\roslyn\src\Compilers\CSharp\csc\csc.csproj...
  Restore completed in 231.29 ms for C:\GitHub\roslyn-internal\roslyn\src\Compilers\CSharp\Portable\CSharpCodeAnalysis.csproj.
  Restore completed in 231.29 ms for C:\GitHub\roslyn-internal\roslyn\src\Compilers\Core\Portable\CodeAnalysis.csproj.
  Restore completed in 548.49 ms for C:\GitHub\roslyn-internal\roslyn\src\Compilers\CSharp\csc\csc.csproj.
  CodeAnalysis -> C:\GitHub\roslyn-internal\roslyn\Binaries\Release\Dlls\CodeAnalysis\Microsoft.CodeAnalysis.dll
  CSharpCodeAnalysis -> C:\GitHub\roslyn-internal\roslyn\Binaries\Release\Dlls\CSharpCodeAnalysis\Microsoft.CodeAnalysis.CSharp.dll
  csc -> C:\GitHub\roslyn-internal\roslyn\Binaries\Release\Exes\csc\netcoreapp2.0\csc.dll
  csc -> C:\GitHub\roslyn-internal\roslyn\Binaries\Release\Exes\csc\net46\csc.exe

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:23.32
// Result = Success

// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet "C:\GitHub\roslyn-internal\roslyn\Binaries/Release/Exes/csc/netcoreapp2.0/csc.dll" -noconfig @repro.rsp
Result  0: 1 op, 4742050516.60 ns, 4.7421 s/op
Result  1: 1 op, 4704319451.16 ns, 4.7043 s/op
Result  2: 1 op, 4700217811.46 ns, 4.7002 s/op
Result  3: 1 op, 4680294251.75 ns, 4.6803 s/op
Result  4: 1 op, 4632986464.10 ns, 4.6330 s/op
Result  5: 1 op, 5249941553.75 ns, 5.2499 s/op
Result  6: 1 op, 5302368986.07 ns, 5.3024 s/op
Result  7: 1 op, 4918268340.75 ns, 4.9183 s/op
Result  8: 1 op, 5010438165.28 ns, 5.0104 s/op
Result  9: 1 op, 5257295731.68 ns, 5.2573 s/op
Result 10: 1 op, 4868777854.53 ns, 4.8688 s/op
Result 11: 1 op, 4740934484.13 ns, 4.7409 s/op
Result 12: 1 op, 4917075437.80 ns, 4.9171 s/op
Result 13: 1 op, 4834390302.47 ns, 4.8344 s/op
Result 14: 1 op, 4848839132.29 ns, 4.8488 s/op
Result 15: 1 op, 4768303549.55 ns, 4.7683 s/op
Result 16: 1 op, 4757527224.67 ns, 4.7575 s/op
Result 17: 1 op, 4618993920.53 ns, 4.6190 s/op
Result 18: 1 op, 5357561992.69 ns, 5.3576 s/op
Result 19: 1 op, 4927464236.71 ns, 4.9275 s/op
Result 20: 1 op, 4826922934.52 ns, 4.8269 s/op
Result 21: 1 op, 4884740821.29 ns, 4.8847 s/op
Result 22: 1 op, 4731102819.91 ns, 4.7311 s/op
Result 23: 1 op, 5262026439.92 ns, 5.2620 s/op
Result 24: 1 op, 5139226548.95 ns, 5.1392 s/op

Mean = 4.9073 s, StdErr = 0.0453 s (0.92%); N = 25, StdDev = 0.2263 s
Min = 4.6190 s, Q1 = 4.7360 s, Median = 4.8488 s, Q3 = 5.0748 s, Max = 5.3576 s
IQR = 0.3388 s, LowerFence = 4.2278 s, UpperFence = 5.5831 s
ConfidenceInterval = [4.7378 s; 5.0768 s] (CI 99.9%), Margin = 0.1695 s (3.45% of Mean)
Skewness = 0.69, Kurtosis = 2.07


// ***** BenchmarkRunner: Finish  *****

// * Export *

// * Detailed results *
PlaceholderBenchmarkRunner.PlaceholderMethod: Job-EUGFVE(Toolchain=Perf.ExternalProcessToolchain, LaunchCount=0, RunStrategy=Monitoring, TargetCount=25, WarmupCount=0) [Commit=HEAD^]
Runtime = ; GC =
Mean = 4.8848 s, StdErr = 0.0464 s (0.95%); N = 25, StdDev = 0.2322 s
Min = 4.6315 s, Q1 = 4.7475 s, Median = 4.8250 s, Q3 = 4.9845 s, Max = 5.8027 s
IQR = 0.2370 s, LowerFence = 4.3920 s, UpperFence = 5.3399 s
ConfidenceInterval = [4.7108 s; 5.0587 s] (CI 99.9%), Margin = 0.1740 s (3.56% of Mean)
Skewness = 2.39, Kurtosis = 9.98


PlaceholderBenchmarkRunner.PlaceholderMethod: Job-EUGFVE(Toolchain=Perf.ExternalProcessToolchain, LaunchCount=0, RunStrategy=Monitoring, TargetCount=25, WarmupCount=0) [Commit=HEAD]
Runtime = ; GC =
Mean = 4.9073 s, StdErr = 0.0453 s (0.92%); N = 25, StdDev = 0.2263 s
Min = 4.6190 s, Q1 = 4.7360 s, Median = 4.8488 s, Q3 = 5.0748 s, Max = 5.3576 s
IQR = 0.3388 s, LowerFence = 4.2278 s, UpperFence = 5.5831 s
ConfidenceInterval = [4.7378 s; 5.0768 s] (CI 99.9%), Margin = 0.1695 s (3.45% of Mean)
Skewness = 0.69, Kurtosis = 2.07


Total time: 00:05:37 (337.99 sec)

// * Summary *

BenchmarkDotNet=v0.10.9, OS=Windows 10.0.16299
Processor=Intel Core i7-7600U CPU 2.80GHz (Kaby Lake), ProcessorCount=4
Frequency=2835939 Hz, Resolution=352.6169 ns, Timer=TSC
.NET Core SDK=2.1.105
  [Host] : .NET Core 2.0.7 (Framework 4.6.26328.01), 64bit RyuJIT DEBUG

Toolchain=Perf.ExternalProcessToolchain  LaunchCount=0  RunStrategy=Monitoring
TargetCount=25  WarmupCount=0

            Method | Commit |    Mean |    Error |   StdDev | Rank |
------------------ |------- |--------:|---------:|---------:|-----:|
 PlaceholderMethod |   HEAD | 4.907 s | 0.1695 s | 0.2263 s |    1 |
 PlaceholderMethod |  HEAD^ | 4.885 s | 0.1740 s | 0.2322 s |    1 |

// * Legends *
  Commit : Value of the 'Commit' parameter
  Mean   : Arithmetic mean of all measurements
  Error  : Half of 99.9% confidence interval
  StdDev : Standard deviation of all measurements
  Rank   : Relative position of current benchmark mean among all benchmarks (Arabic style)
  1 s    : 1 Second (1 sec)

// ***** BenchmarkRunner: End *****
// * Artifacts cleanup *
</details>

**Comment by CyrusNajmabadi** [comment]
@agocke can you help interpret this?  It looks like the diffs are well within the margin of error.  yes?

File: src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs [hunk 2]
```diff
@@ -433,21 +431,31 @@ public override SingleNamespaceOrTypeDeclaration VisitEnumDeclaration(EnumDeclar
                 diagnostics: diagnostics.ToReadOnlyAndFree());
         }
 
-        private static string[] GetEnumMemberNames(SeparatedSyntaxList<EnumMemberDeclarationSyntax> members, ref SingleTypeDeclaration.TypeDeclarationFlags declFlags)
+        private static readonly ObjectPool<ImmutableHashSet<string>.Builder> s_memberNameBuilderPool =
+            new ObjectPool<ImmutableHashSet<string>.Builder>(() => ImmutableHashSet.CreateBuilder<string>());
+
+        private static ImmutableHashSet<string> ToImmutableAndFree(ImmutableHashSet<string>.Builder builder)
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs]: Yup yup :)

**Comment by CyrusNajmabadi** [comment]
@gafter can you take a look?  The dependent PR has gone in now.

**Comment by CyrusNajmabadi** [comment]
@agocke Can you confirm my reading that this means there is no appreciable perf impact to this change?  @gafter @jcouv could you take a look at this rather simple change.  Note: actually testing this on Roslyn brought down the time to test lookup from over 7 seconds to an almost 0.2 seconds.  That's a big win! :)  Hash lookups much better than a linear scan of every top level symbol name in the entire solution

**Comment by CyrusNajmabadi** [comment]
Note: this search function is used both by add-using and fully-qualify (though only for the direct project you start in, and not for searching other projects in the solution), as well as the public SymbolFinder.FindDeclarationsAsync entrypoint.  So improvements here will def have a benefit.

It is *not* currently used by FindRefs, or goto-impl/override.  Though it's possible searches could be sped up by using these helpers before having to build/examine our own IDE indexes here.  Though investigations would hav eto be done to know for sure.

**Comment by agocke** [comment]
@CyrusNajmabadi The "rank" part of the output is the most important here. That both are rank "1" indicates that there was no statistically significant difference in end-to-end performance of the compiler between the two commits.

**Comment by agocke** [comment]
If you have other perf measurements I think you should also post those benchmarks. Without some definite perf improvement in a specific scenario, I wouldn't take this change.

**Comment by CyrusNajmabadi** [comment]
@agocke i mentioned them above.  

Note: i was the person who added this API in the early days of Roslyn *precisely* for speeding up add-using and for other IDE search tasks.  It looks like it was changed at one point to not be a set, without understanding that scenario, and also making it so we expose mutable data.

To me, it was a mistake to have made that change, and this is simply rectifying things (both fixing the exposing of mutable data, and improving perf).  I'm not sure hte best way for us to prevent these sorts of regressions in the future.  At one point there was a roslyn perf harness, but i'm not sure the current state of it.

**Comment by agocke** [comment]
My work is kind of the replacement for that, but it only measures end-to-end perf, which this change is too small to affect. Could you post detailed results for a benchmark of the API change, then? Means, standard deviations, maybe a t-test for significance?

**Comment by CyrusNajmabadi** [comment]
Do you have a good way to collect that data?  All i did was make a scenairo and run it several hundred times.  the results were consistent every time i tried to do this.

**Comment by agocke** [comment]
https://github.com/dotnet/BenchmarkDotNet can give you that information. They have pretty good documentation for writing tests, as well.

**Comment by CyrusNajmabadi** [comment]
Ok.  I had some trouble using htat directly.  But just doing the benchmarking manually was no problem.  Here are the results.

Before my change:

Raw run times.  Note, these are in milliseconds.  

```
11380
5980
5843
6852
5913
5896
6235
5914
5995
5879
6038
5905
5873
5869
5918
5870
6023
5883
5900
6597
6063
5874
5925
5899
5879
```

This gives the following values:


avg(ms) | stddevp(ms) | error(ms)
-- | -- | --
6,216.12 | 1079.289593 | 215.8579186

After my change:

```
679
917
1070
835
981
729
402
403
400
386
392
390
408
398
394
389
391
402
397
380
409
377
420
452
492
```

avg(ms) | stddevp(ms) | error(ms)
-- | -- | --
515.72 | 210.4803117 | 42.09606233

This is roughly a 12x improvement.

File: src/Compilers/VisualBasic/Portable/Declarations/DeclarationTable.vb [hunk 3]
```diff
@@ -285,11 +285,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
...
             cancellationToken As CancellationToken) As Boolean
 
-            ' note: because we want to do case-insensitive lookup, and because our decl tables
-            ' do not store members in a way that can lookup in a case-insensitive manner, we have
-            ' to defer to just calling the normal predicate version.
-            Return ContainsName(
-                mergedRoot, Function(n) IdentifierComparison.Equals(n, name), filter, cancellationToken)
+            Return ContainsNameHelper(
+                mergedRoot,
+                Function(n) IdentifierComparison.Equals(n, name),
```
Comment (agocke) [inline src/Compilers/VisualBasic/Portable/Declarations/DeclarationTable.vb]: Consider passing through `name` so you don't allocate in these lambdas.

Comment (CyrusNajmabadi) [reply src/Compilers/VisualBasic/Portable/Declarations/DeclarationTable.vb]: it's just a single allocation, so it's not too bad :)

**Comment by CyrusNajmabadi** [comment]
@agocke are you ok merging in?

**Pull request merged**